go-controller/pkg/ovn/policy.go

```diff
@@ -276,7 +278,12 @@ func (oc *Controller) createMulticastAllowPolicy(ns string, nsInfo *namespaceInf
 	}
 
 	// Add all ports from this namespace to the multicast allow group.
-	for _, portName := range nsInfo.addressSet {
+	pods, err := oc.watchFactory.GetPods(ns)
```

Comment:
potential race condition? (I'm not sure)

Reply:
add logical port will call  addPodToNamespace, which will call, podAddAllowMulticastPolicy, which will add the port to the port group. I think there probably is a race condition here if pods are being added while this policy is getting created. I don't see locking in the current code. If we refactor this to use real port group objects with proper locking around it then it will solve the problem. Maybe we just need to add a comment here for now saying // FIXME, potential race

Reply:
@trozet @danwinship the real fix here is to always create the namespace port group; @JacobTanenbaum wants this for egressfirewall too. But that is an increase in OVN resources for something that eg @girishmg doesn't care about, which is why we made it optional. But having it optional (and not storing port names in address sets anymore) is exactly what makes the race condition possible. I can add some locking though.

Reply:
@danwinship @trozet actually we're clear. All codepaths that call `podAddAllowMulticastPolicy()`, `podDeleteAllowMulticastPolicy()`, or `createDefaultDenyMulticastPolicy()` hold the namespaceInfo object lock.